src/TemplatesBuilder.js

```diff
@@ -129,8 +129,11 @@ export default class TemplatesBuilder {
     propertyTemplate.defaults = this.defaultsForLiteral(propertyTerm)
     propertyTemplate.validationRegex =
       this.validationRegexForLiteral(propertyTerm)
-    propertyTemplate.validationDataType =
-      this.validationDataTypeForLiteral(propertyTerm)
+    if (this.validationDataTypeForLiteral(propertyTerm)) {
```

Comment:
Can you assign `this.validationDataTypeForLiteral(propertyTerm)` to a variable instead of invoking twice?